### ISSUE TITLE
Cypress component tests report eslint erorrs

### ISSUE THREAD

**Issue by braughtg** [DickinsonCollege/FarmData2#600]
When running the cypress component tests lots of eslint errors are reported. These appear to be a configuration issue rather than an actual issue with the code as they mostly report undefined variables or functions in cypress (e.g. 'cy' is not defined).  But the tests themselves run perfectly fine.

**Steps to Reproduce**
1. `cd` to farmdata2_modules
2. run `./test_runner.bash ct --spec **/Error*.spec.comp.js`
3. observe the errors at the top of the ouput.

Also note that the test itself passes.  This suggests maybe that eslint is running in a different context than the cypress tests themselves. 

Simply disabling the running of eslint before the component tests would be an acceptable solution.

**Comment by FutzMonitor** [comment]
I had to run this inside of the `/FarmData2/farmdata2` directory since the project has been reorganized since the creation of this issue. 

This is the result of the output: 
![image](https://user-images.githubusercontent.com/31524934/222781601-cf4b1ff0-c47d-44d1-a55f-6182a957cb76.png)

**Comment by braughtg** [comment]
Interesting I get the following:
![image](https://user-images.githubusercontent.com/16962482/222814744-5f1a78a1-55cf-4b07-98c4-2640d52af3f7.png)

Is it possible that this is related to your work on #601 where you may have removed the ErrorBannerComponent?

Try the following command instead... It will generate similar errors:
- `./test_runner.bash ct --spec **/Date*.spec.comp.js`

**Comment by FutzMonitor** [comment]
> Is it possible that this is related to your work on #601 where you may have removed the ErrorBannerComponent?
> 
> Try the following command instead... It will generate similar errors:
> 
>     * `./test_runner.bash ct --spec **/Date*.spec.comp.js`

I ran the above command on my laptop using the main branch which still has the `ErrorBannerComponent` present. I re-ran the command above on my desktop computer when I realized there was a mistake in my previous comment. I forgot the add the forward slash before the word "Error". Correcting this mistake produces the same result you had: 

![image](https://user-images.githubusercontent.com/31524934/222823768-1189df80-2c17-4ce3-a2c4-a19241d33236.png)

Running the `./test_runner.bash ct --spec **/Date*.spec.comp.js` produces the following output: 

![image](https://user-images.githubusercontent.com/31524934/222824272-b7559e07-d967-4427-936e-e5f72d00a2cb.png)

**Comment by FutzMonitor** [comment]
I can try taking a look into this issue.

**Comment by FutzMonitor** [comment]
I'm taking a look at the eslint documentation and there's a page on disabling eslint for all JavaScript files [[link](https://eslint.org/docs/latest/use/configure/ignore)]. However, this solution isn't working. The error continue to occur. Alternatively, I could add the lines `/* eslint-disable */` to the top of every single `**.spec.comp.js` along with the component files themselves. I think this is a pretty unsatisfactory solution however. It would be a lot nicer to simply disable eslint in one location instead of having to worry about multiple location. Do you have any tips of some other way to disable the eslint?

**Comment by braughtg** [comment]
I don't think disabling es-lint in the files is the right solution.  The plan eventually will be to have es-lint running in VSCodium and enforcing a style on all of the code so that it is uniform across the project.  If it is possible to disable this just for the running of the tests via cypress in the test container that would probably be fine.  

Maybe there is a flag that you can pass to the Cypress test runner?  Then that could be added to the call in the `test_runner.bash` script?  Another alternative, is that maybe something else needs to be installed into the `fd2test` project that is used to run the tests in the dev container?

I haven't had much time to investigate this.

**Comment by FutzMonitor** [comment]
Where exactly is the eslint installed? I'm looking around the project and the fd2dev's docker files, and I can't quite pinpoint exactly where eslint is being introduced.

**Comment by braughtg** [comment]
I think it must be being installed as part of the normal cypress instal in the dev container. I think I would start by looking at how to run cypress from the command line and then looking at the `test_runner.bash` script.  That will give you a feel for how this comes together.

Note that the `dockerfile` for the `dev` container also creates the `fd2test` project in the `fd2dev` home directory.  That is really where the testing is running from.  The `farmdata2` directory is mounted into that `fd2test` directory to make all of the `.spec` files available to Cypress.

**Comment by FutzMonitor** [comment]
I found it within the `fd2test` project inside the `fd2dev` container. I'll see what I can do to disable `eslint` just for Cypress testing.

![image](https://user-images.githubusercontent.com/31524934/223470259-9f0f6480-35e9-4d89-9c79-07134d8d9410.png)

**Comment by FutzMonitor** [comment]
It was mentioned that instead of running the `RUN` command in order to create the `.eslintignore` file and its contents, it would be more consistent to make use of the `COPY` command. However, I'm reading documentation on this command [[source](https://www.geeksforgeeks.org/docker-copy-instruction)] and it seems to copy pre-existing files on the local machine into the project. Does that mean I should make the `.eslintignore` a file in the project and simple have the `.gitignore` not make it appear in the repository? Having it available on the local machine is useful because people who attempt to run Cypress tests via the command line on their host machine versus the dev container will experience the very issues I'm attempting to resolve.

**Comment by braughtg** [comment]
I would suggest creating a file in `docker/dev` called `eslintignore` that contains the content that you want `.eslintignore` in the container to have.  Then add a line to copy this fie into the container to the `Dockerfile` near where it uses `COPY` to copy the other files from `docker/dev` (lines 169-182).  Notice that those lines copy the files and also rename them to include the `.` at the start.  This is so that these files are not hidden files in the repo (no `.`) but are hidden files, as they are supposed to be, in the container.

**Comment by FutzMonitor** [comment]
I noticed that the `webpack-dev-server` runs some linting tests as well. So it might be worthwhile exploring if the linting tests are happening there and being pipelined with the cypress tests.

**Comment by braughtg** [comment]
> I noticed that the `webpack-dev-server` runs some linting tests as well.

Sounds promising!

**Comment by braughtg** [comment]
> I uninstalled webpack and ran the test runner, but it worked as if nothing was uninstalled.

Are you saying here that the linting errors were also reported when you ran the tests after uninstalling webpack?

**Comment by FutzMonitor** [comment]
> Are you saying here that the linting errors were also reported when you ran the tests after uninstalling webpack?

Yes, the linting errors were reported despite webpack being uninstalled.
